src/vdi.ps1

```diff
@@ -95,74 +130,69 @@ function Get-XoVdi {
         [Parameter(ParameterSetName = "All")]
         [Parameter(ParameterSetName = "SrFilter")]
         [Parameter(ParameterSetName = "Filter")]
-        [int]$Limit
+        [int]$Limit = $(if ($null -ne $script:XO_DEFAULT_LIMIT) { $script:XO_DEFAULT_LIMIT } else { 25 })
     )
     
-    begin {
-        Write-Verbose "Getting VDIs from XO"
-        
-        # Base parameters for API requests
-        $params = Remove-XoEmptyValues @{
-            fields = $script:XO_VDI_FIELDS
-            filter = if ($PSCmdlet.ParameterSetName -eq "SrFilter") { "sr:$SrUuid" } elseif ($PSCmdlet.ParameterSetName -eq "Filter") { $Filter } else { $null }
-            limit = $Limit
-        }
+    if (-not $script:XoHost -or -not $script:XoRestParameters) {
+        throw "Not connected to Xen Orchestra. Call Connect-XoSession first."
     }
     
-    process {
-        # Handle individual VDI retrieval
-        if ($PSCmdlet.ParameterSetName -eq "VdiId") {
-            foreach ($id in $VdiId) {
-                try {
-                    Write-Verbose "Getting VDI with ID $id"
-                    $vdi = Invoke-RestMethod -Uri "$script:XoHost/rest/v0/vdis/$id" @script:XoRestParameters -Body $params
-                    
-                    if ($vdi) {
-                        ConvertTo-XoVdiObject $vdi
-                    }
-                    else {
-                        Write-Warning "No VDI found with ID $id"
-                    }
-                }
-                catch {
-                    Write-Error "Failed to retrieve VDI with ID $id. Error: $_"
-                }
-            }
+    $params = @{
+        fields = $script:XO_VDI_FIELDS
+    }
+    
+    if ($PSCmdlet.ParameterSetName -eq "SrFilter") {
+        $params['filter'] = "sr:$SrUuid"
+    } elseif ($PSCmdlet.ParameterSetName -eq "Filter") {
+        $params['filter'] = $Filter
+    }
+    
+    if ($Limit -ne 0 -and $PSCmdlet.ParameterSetName -ne "VdiId") {
+        $params['limit'] = $Limit
+        if (!$PSBoundParameters.ContainsKey('Limit')) {
+            Write-Warning "No limit specified. Using default limit of $Limit. Use -Limit 0 for unlimited results."
+        }
+    }
+
+    if ($PSCmdlet.ParameterSetName -eq "VdiId") {
```

Comment:
Why do some object types use XxxId for UUID parameters and some other use XxxUuid?